src/os_win/os_snprintf.c

```diff
@@ -8,17 +8,47 @@
 
 #include "wt_internal.h"
 
-_Check_return_opt_ int __cdecl _wt_snprintf(
-    _Out_writes_(_MaxCount) char * _DstBuf,
-    _In_ size_t _MaxCount,
-    _In_z_ _Printf_format_string_ const char * _Format, ...)
+/*
+ * __wt_vsnprintf_len_incr --
+ *	POSIX vsnprintf convenience function, incrementing the returned size.
+ */
+int
+__wt_vsnprintf_len_incr(
+    char *buf, size_t size, size_t *retsizep, const char *fmt, va_list ap)
 {
-	va_list args;
-	WT_DECL_RET;
+	int len;
+
+	/*
+	 * WiredTiger calls with length 0 to get the needed buffer size. Call
+	 * the count only version in this case, _vsnprintf_s will invoke the
+	 * invalid parameter handler if count is less than or equal to zero.
+	 */
+	if (size == 0) {
+		*retsizep += (size_t)_vscprintf(fmt, ap) + 1;
+		return (0);
+	}
+
+	/*
+	 * Additionally, the invalid parameter handler is invoked if buffer or
+	 * format is a NULL pointer.
+	 */
+	if (buf == NULL || fmt == NULL)
+		return (EINVAL);
+
+	/*
+	 * If the storage required to store the data and a terminating null
+	 * exceeds size, the invalid parameter handler is invoked, unless
+	 * count is _TRUNCATE, in which case as much of the string as will
+	 * fit in the buffer is written and -1 returned.
+	 */
+	if ((len = _vsnprintf_s(buf, size, _TRUNCATE, fmt, ap)) >= 0) {
+		*retsizep += (size_t)len;
+		return (0);
+	}
 
-	va_start(args, _Format);
-	ret = _wt_vsnprintf(_DstBuf, _MaxCount, _Format, args);
-	va_end(args);
+	/* Return the buffer size required. */
+	if (len == -1)
+		*retsizep += (size_t)_vscprintf(fmt, ap) + 1;
```

Comment:
Thanks, @sulabhM, I think you're correct. (The reason the `+1` was there was because it was in the original code and I just copied that code.)

I've pushed a fix in 72ca3ef.